ws-server/src/server.rs

```diff
@@ -225,10 +229,9 @@ enum HandshakeResponse<'a, M> {
 	},
 }
 
-async fn handshake<M>(socket: tokio::net::TcpStream, mode: HandshakeResponse<'_, M>) -> Result<(), Error>
-where
-	M: Middleware,
-{
+async fn handshake<M: Middleware>(socket: tokio::net::TcpStream, mode: HandshakeResponse<'_, M>) -> Result<(), Error> {
```

Comment:
Probably out of scope of this PR but it feels like having an `accept_handshake` and `reject_handshake` call (or similar) would make more sense, since this call basically just switches on an enum and has very little shared between them?